File: Backend/src/main/java/lan/dk/podcastserver/business/stats/StatsPodcastType.java
package lan.dk.podcastserver.business.stats;

import com.fasterxml.jackson.annotation.JsonIgnore;
import javaslang.collection.Set;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * Created by kevin on 28/04/15 for HackerRank problem
 */
@Getter
@AllArgsConstructor
public class StatsPodcastType {

    private final String type;
    private final Set<NumberOfItemByDateWrapper> values;

    @JsonIgnore
    public Boolean isEmpty() {
        return values.isEmpty();
    }
}

File: backend/src/main/kotlin/com/github/davinkevin/podcastserver/business/stats/StatsPodcastType.kt
package com.github.davinkevin.podcastserver.business.stats

import com.fasterxml.jackson.annotation.JsonIgnore

/**
 * Created by kevin on 28/04/15 for HackerRank problem
 */
data class StatsPodcastType(val type: String, val values: Set<NumberOfItemByDateWrapper> = setOf()) {

    val isEmpty: Boolean
        @JsonIgnore get() = values.isEmpty()
}
